src/providers/Web3Provider.tsx

```diff
@@ -0,0 +1,52 @@
+"use client";
+
+import { WagmiProvider, createConfig, http } from "wagmi";
+import {
+  arbitrum,
+  base,
+  mainnet,
+  optimism,
+  polygon,
+  sepolia,
+} from "wagmi/chains";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ConnectKitProvider, getDefaultConfig } from "connectkit";
+import type { ReactNode } from "react";
+
+const config = createConfig(
+  getDefaultConfig({
+    // Your dApps chains
+    chains: [mainnet, optimism, arbitrum, base, polygon, sepolia],
+    transports: {
+      [mainnet.id]: http(process.env.NEXT_PUBLIC_MAINNET_RPC_URL ?? ""),
+      [optimism.id]: http(process.env.NEXT_PUBLIC_OPTIMISM_RPC_URL ?? ""),
+      [arbitrum.id]: http(process.env.NEXT_PUBLIC_ARBITRUM_RPC_URL ?? ""),
+      [base.id]: http(process.env.NEXT_PUBLIC_BASE_RPC_URL ?? ""),
+      [polygon.id]: http(process.env.NEXT_PUBLIC_POLYGON_RPC_URL ?? ""),
+      [sepolia.id]: http(process.env.NEXT_PUBLIC_SEPOLIA_RPC_URL ?? ""),
+    },
```

Comment:
`#must` ask Candide what they'd prefer here. Also, we don't need to have these as env variables unless they provide us with a set of private RPCs which I doubt they will.